src/pymodaq/utils/parameter/pymodaq_ptypes/itemselect.py

```diff
@@ -36,9 +41,13 @@ def get_value(self):
             Returns
             -------
             dictionnary
-                The dictionnary of all_items compared to the slelectedItems.
+                The dictionnary of all_items compared to the selectedItems.                                
         """
-        selitems = [item.text() for item in self.selectedItems()]
+        if self.hasCheckbox:            
+            selitems = [item.text() for item in self.all_items() if item.checkState()!=0]            
```

Comment:
What happens when you click on the item (as we did previously)? Does the check box becomes checked? Or is it possible to be selected with a checkbox not checked?

Reply:
When you check the box, is the item put in the selected list?

Reply:
I'd like you to write tests to verify this behaviour please

Reply:
It does nothing when you click, the selection is decided with the item.checkState() which is just looking for the checkbox value for each item
0 == unchecked
1 == somewhat check (used with tristate)
2 == check

This function is as simple as **self.all_items()**  and is built similarly as **allitems = [item.text() for item in self.all_items()]** 
 A test looks a bit overkilled honestly, the parameter_ex.py is already testing all of what you are asking.

Where would you like to put it?

Reply:
> When you check the box, is the item put in the selected list?

Yes the ordering depends on the initial order as it goes through all the list from up to down

Reply:
> What happens when you click on the item (as we did previously)? Does the check box becomes checked? Or is it possible to be selected with a checkbox not checked?

- It does nothing, the signal is related to the itemChanged now which will only be modified if the text or the checkbox is changed 
- It is not possible to select without checking the checkbox, only ticking the checkbox will emit a signal

Reply:
> A test looks a bit overkilled honestly, the parameter_ex.py is already testing all of what you are asking.

Well except no one is going to run the example when commiting changes of behaviour (such as what you do right now). This is why programmatic tests are needed, I would put them here:
![image](https://github.com/PyMoDAQ/PyMoDAQ/assets/6609955/8aeb5eec-5e65-4a29-8ba8-6420e7653929)

And we have to make sure that both objects behave exactly the same (in terms of selection and ordering) otherwise the behaviour would change is we go from using check boxes to not using them